packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/object/services/CsvObjectExporter.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.object.services;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.core.api.IIdentityService;
+import org.eclipse.sirius.components.core.api.ILabelService;
+import org.eclipse.sirius.web.application.object.services.api.IObjectExporter;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to export objects as CSV resources.
+ *
+ * @author gdaniel
+ */
+@Service
+public class CsvObjectExporter implements IObjectExporter {
+
+    private final IIdentityService identityService;
+
+    private final ILabelService labelService;
+
+    public CsvObjectExporter(IIdentityService identityService, ILabelService labelService) {
+        this.identityService = Objects.requireNonNull(identityService);
+        this.labelService = Objects.requireNonNull(labelService);
+    }
+
+
+    @Override
+    public boolean canHandle(List<Object> objects, String mediaType) {
+        return "text/csv".equals(mediaType);
+    }
+
+    @Override
+    public Optional<byte[]> getBytes(List<Object> objects, String mediaType) {
+        String headers = "id,type,label" + System.lineSeparator();
```

Comment:
Not sure you want the headers in the generated CSV file.
I put them because I feel it is easier to understand the content, but I can remove that if you don't want them.